app/src/App.spec.ts

```diff
@@ -30,4 +39,46 @@ describe("App", () => {
             expect(getAccessTokenSilently).toHaveBeenCalledOnce();
         });
     });
+
+    it("shows the banner when offline", async () => {
+        const addNotification = vi.fn();
+
+        vi.spyOn(isConnected, "value", "get").mockReturnValue(false);
+        vi.spyOn(useNotificationStore(), "addNotification").mockImplementation(addNotification);
+
+        mount(App, {
+            shallow: true,
+        });
+
+        // Wait for the setTimeout and watch to trigger
+        await waitForExpect(() => {
+            expect(addNotification).toHaveBeenCalledWith({
+                id: "offlineBanner",
+                title: "You are offline",
+                description:
+                    "You can still use the app and browse through offline content, but some content (like videos) might not be available.",
+                state: "warning",
+                type: "banner",
+                icon: SignalSlashIcon,
+            });
+        }, 1000);
+    });
+
+    it("doesnt show the banner when online", async () => {
+        const removeNotification = vi.fn();
+
+        vi.spyOn(isConnected, "value", "get").mockReturnValue(true);
+        vi.spyOn(useNotificationStore(), "removeNotification").mockImplementation(
+            removeNotification,
+        );
+
+        mount(App, {
+            shallow: true,
+        });
+
+        // Wait for the setTimeout and watch to trigger
+        await waitForExpect(() => {
+            expect(removeNotification).toHaveBeenCalledWith("offlineBanner");
+        }, 1000);
+    });
```

Comment:
Thanks for adding tests! Just a small note, there's no need to mock the store here I think. Generally stores work really well in unit tests. You can just directly use it and do assertions on it.

```ts
        const notificationStore = useNotificationStore();
        expect(notificationStore.addNotification).toHaveBeenCalledWith(
            expect.objectContaining({ state: "error" }),
        );
```

I copied this from a test in the CMS, `SettingsPage.spec.ts`. Also notice the `expect.objectContaining` here by the way, this is a very handy method that you can use to only check part of an object, instead of having to check for all fields.